Signal/src/util/OWSBackupExportJob.m

```diff
@@ -149,35 +308,32 @@ - (void)start
         [self updateProgressWithDescription:NSLocalizedString(@"BACKUP_EXPORT_PHASE_EXPORT",
                                                 @"Indicates that the backup export data is being exported.")
                                    progress:nil];
-        [self exportDatabase:^(BOOL exportDatabaseSuccess) {
-            if (!exportDatabaseSuccess) {
-                [self failWithErrorDescription:
-                          NSLocalizedString(@"BACKUP_EXPORT_ERROR_COULD_NOT_EXPORT",
-                              @"Error indicating the a backup export could not export the user's data.")];
-                return;
-            }
-
-            if (self.isComplete) {
+        if (![self exportDatabase]) {
+            [self failWithErrorDescription:
+                      NSLocalizedString(@"BACKUP_EXPORT_ERROR_COULD_NOT_EXPORT",
+                          @"Error indicating the a backup export could not export the user's data.")];
+            return;
+        }
+        if (self.isComplete) {
+            return;
+        }
+        [self saveToCloudWithCompletion:^(NSError *_Nullable saveError) {
+            if (saveError) {
+                [weakSelf failWithError:saveError];
                 return;
             }
-            [self saveToCloud:^(NSError *_Nullable saveError) {
-                if (saveError) {
-                    [weakSelf failWithError:saveError];
+            [self cleanUpCloudWithCompletion:^(NSError *_Nullable cleanUpError) {
```

Comment:
Yup, I finally made the "with completion" change.

Reply:
If you send an emoji to someone as old as me, you're running an 80% chance of misinterpretation.  This is presumably the "water evaporating from wet hair after shower" emoji.

Reply:
🚜 🤒 🇫🇰